### DISCUSSION TITLE
Hungarian matching: `ValueError: matrix contains invalid numeric entries`

### DISCUSSION THREAD

**Discussion by juliagorman** [talmolab/sleap#1321] [Help!] [answered]
I have an experiment where there is only ever one animal in the enclosure, however, I have found that ithe performance is better when I train as a multianimal top down instead of the single animal. This however usually means I have extra skeletons in the prediction. I recently tried to use the pre-tracker cleaning function to let sleap know that I only have one animal, I will attach of photo of what i did. I then got an error from trying to run that inference, and this was the resulting error on the command line. Is there no easy way to do what I am trying to do? 
<img width="1211" alt="Screenshot 2023-03-16 at 8 39 09 AM" src="https://user-images.githubusercontent.com/32439305/225932412-6b09ccbb-72b3-4e1a-b149-2e35c17f72a2.png">
<img width="1044" alt="Screenshot 2023-03-16 at 8 38 45 AM" src="https://user-images.githubusercontent.com/32439305/22593
<img width="971" alt="Screenshot 2023-03-16 at 8 38 19 AM" src="https://user-images.githubusercontent.com/32439305/225932456-b6cd62bf-dab6-456a-a033-b84b9cd4f0da.png">
2443-ceedf74f-0f3c-4f3a-a19d-e10d1fcd98d1.png">

**Comment by juliagorman** [comment]
here is the photo that didnt attach properly
<img width="1044" alt="Screenshot 2023-03-16 at 8 38 45 AM" src="https://user-images.githubusercontent.com/32439305/225932856-e4985200-96e2-4693-93d2-d455136d9e66.png">

**Comment by laurelrr** [comment, marked as answer]
I think I have encountered something similar.  Try setting tracker to 'flow' and the matching method to 'greedy' instead of hungarian.  See if that helps.

**Comment by roomrys** [comment]
Hi @juliagorman,

Just as @laurelrr stated, this error will occasionally pop-up if the cost matrix to be solved by the hungarian method contains invalid entries (an unsolvable matrix). Switching over to greedy matching will work perfectly. No need to do a flow tracker unless you want to.

Thanks,
Liezl

**Comment by juliagorman** [comment]
Okay that worked in the sense that I was able to run it, however, now I have 141 difference track identities across the entire video for the same animal. Is there a way to specify that this is the same animal throughout the video so that there is only one track identity?
<img width="1273" alt="Screenshot 2023-03-20 at 8 51 16 AM" src="https://user-images.githubusercontent.com/32439305/226395542-b602eb03-68ba-41d7-bd60-19ac88773d5d.png">

**Comment by roomrys** [comment]
Hi @juliagorman,

Yes! I am glad you specified that it was a single-animal project. You can use a tracker method to cull to the number of target instances (in the case that you are using a multi-animal pose-estimation method - i.e. top-down or bottom-up), but then go ahead and delete all the tracks.
![image](https://user-images.githubusercontent.com/38435167/226461791-b01b386b-a676-4862-9704-6a88a60db253.png)

If there is only a single animal in all the frames, then SLEAP assumes that it is the same animal throughout and will export the analysis file accordingly.

Thanks for posting on the discussion. As always, let us know if this helps!

Thanks,
Liezl
